distributed/worker.py

```diff
@@ -1647,8 +1653,7 @@ def update_data(
         report: bool = True,
         stimulus_id: str = None,
     ) -> dict[str, Any]:
-        if stimulus_id is None:
-            stimulus_id = f"update-data-{time()}"
+        stimulus_id = stimulus_id or f"update-data-{time()}"
```

Comment:
I think this re-expression of the previous code exists only to satisfy the type-checker. The default type is `stimulus_id: str` throughout `worker.py`, but the above handler supports `stimulus_id: Optional[str]` because it's a kwarg.

Reply:
I think previously we introduced this as nullable because the other end never provided a stim ID. This changed now and we should be able to move the stimulus_id to be a required argument. 

Even though we're not enforcing this, I would argue that all of our handlers shuold define kwarg-only signatures but here we are...

This is OK the way it is for now